### PR TITLE
Osher/mgmt 1038 - Should not allow to update hosts when the installation has started

### PR DESCRIPTION
**Pull request by oshercc**
MGMT-1038 prevent update while cluster is installing

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -656,6 +656,12 @@ func (b *bareMetalInventory) UpdateCluster(ctx context.Context, params installer
 		tx.Rollback()
 		return installer.NewUpdateClusterNotFound().WithPayload(common.GenerateError(http.StatusNotFound, err))
 	}
+	if b.clusterApi.IsClusterInstallationInProgress(&cluster) {
```
Comment (avishayt) [inline internal/bminventory/inventory.go:659]:
```suggestion
	if b.clusterApi.HasClusterInstallationStarted(&cluster) {
```

And check if it's installing-in-progress, installing, installed, or error.  User shouldn't be allowed to change the cluster in these states.

Comment (oshercc) [reply internal/bminventory/inventory.go:659]: from the ticket, I understand that the only in installing state nothing can be updated, no ?

**Comment by avishayt** [comment]
The ticket says "after starting the installation", which is all of the states that I mentioned.  Also note that this relates to MGMT-923.

FYI @filanov

**Comment by filanov** [comment]
> The ticket says "after starting the installation", which is all of the states that I mentioned. Also note that this relates to MGMT-923.
> 
> FYI @filanov

Agree with @avishayt

**Comment by oshercc** [comment]
@avishayt @filanov   What about cluster in error ?

**Comment by filanov** [comment]
> @avishayt @filanov What about cluster in error ?

same, can't update

File: internal/bminventory/inventory.go (hunk 1 above)
Comment (ronniel1) [reply internal/bminventory/inventory.go:659]: @oshercc What @avishayt said :)

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -658,6 +658,12 @@ func (b *bareMetalInventory) UpdateCluster(ctx context.Context, params installer
 		tx.Rollback()
 		return installer.NewUpdateClusterNotFound().WithPayload(common.GenerateError(http.StatusNotFound, err))
 	}
+
+	if err = b.clusterApi.VerifyClusterUpdatability(&cluster); err != nil {
```
Comment (filanov) [inline internal/bminventory/inventory.go:662]: rollback

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -658,6 +658,13 @@ func (b *bareMetalInventory) UpdateCluster(ctx context.Context, params installer
...
+		log.WithError(err).Errorf("cluster %s can't be updated in current state", params.ClusterID)
+		tx.Rollback()
```
Comment (filanov) [inline internal/bminventory/inventory.go:664]: if you rebased above master you will not need this line. i added you the review.

**Comment by oshercc** [comment]
@filanov can you please re-approve, had to push conflict fixes.

**Comment by filanov** [comment]
> @filanov can you please re-approve, had to push conflict fixes.

done

**Pull request merged**